pyproject.toml

```diff
@@ -1,14 +1,148 @@
+        'jsonschema>=4.18',  # New reference resolution API
+        'numpy>=1.26.0',
+        'lark',
+        'pandas',
+        'python-dateutil',
+        'pyyaml',
+        'rasterio>=1.3.11',  # Deleted Numpy version in 1.3.10
+        'ruamel.yaml',
+        'sqlalchemy>=2.0',  # GeoAlchemy2 requires >=1.4.
+        'GeoAlchemy2',
+        "alembic",
+        'toolz',
+        'xarray>=0.9',  # >0.9 fixes most problems with `crs` attributes being lost
+        'packaging',
+        'odc-geo>=0.4.8',
+        'deprecat',
+]
+
+[project.urls]
+Homepage = "https://opendatacube.org/"
+Documentation = "https://datacube-core.readthedocs.io/"
+Source = "https://github.com/opendatacube/datacube-core/"
+Issues = "https://github.com/opendatacube/datacube-core/issues"
+Chat = "https://discord.com/invite/4hhBQVas5U"
+
+[project.optional-dependencies]
+performance = ['ciso8601', 'bottleneck']
+distributed = ['distributed', 'dask[distributed]']
+s3 = ['boto3', 'botocore']
+cf = ['compliance-checker>=4.0.0']
+netcdf = ['netcdf4']
+postgres = ['psycopg2']
+
+[dependency-groups]
+dev = [
+    # Test requirements
+    'hypothesis',
+    'pytest',
+    'pytest-cov',
+    'pytest-timeout',
+    'pytest-httpserver',
+    'moto<5.0',  # 5.0 will require changes to some tests.
+    'psycopg2',
+    'netcdf4',
+
+    # Linting tools
+    'pycodestyle',
+    'pylint',
+
+    # Types and type checking
+    "mypy",
+    "types-affine",
+    "types-cachetools",
+    "types-jsonschema",
+    "types-psutil",
+    "types-python-dateutil",
+    "types-PyYAML",
+    "types-redis",
+    "types-setuptools",
+    "boto3-stubs",
+    "botocore-stubs"
+]
+doc = [
+    'Sphinx',
+    'sphinx_autodoc_typehints',  # Propagate mypy info into docs
+    'sphinx-click',
+    'recommonmark',
+    'autodocsumm',
+    'beautifulsoup4',
+    'nbsphinx',
+    'ipython',
+    'pydata-sphinx-theme',
+    'sphinx-design',
+]
+
+[project.scripts]
+datacube = 'datacube.scripts.cli_app:cli'
+datacube-search = 'datacube.scripts.search_tool:cli'
+datacube-worker = 'datacube.execution.worker:main'
+
+[build-system]
+requires = ["setuptools>=69", "setuptools_scm[toml]>=3.4"]
+# v61 Add support for pyproject.toml, and include-package-data defaults to try when using pyproject.toml
+# v69 includes py.typed files by default
+build-backend = "setuptools.build_meta"
+
+[project.entry-points."datacube.plugins.io.read"]
+netcdf = "datacube.drivers.netcdf.driver:reader_driver_init"
+
+[project.entry-points."datacube.plugins.io.write"]
+netcdf = 'datacube.drivers.netcdf.driver:writer_driver_init'
+
+[project.entry-points."datacube.plugins.index"]
+postgres = 'datacube.index.postgres.index:index_driver_init'
+null = 'datacube.index.null.index:index_driver_init'
+memory = 'datacube.index.memory.index:index_driver_init'
+postgis = 'datacube.index.postgis.index:index_driver_init'
+
+[tool.setuptools.packages.find]
+exclude = ['integration_tests', 'tests', 'examples', 'docs']
+include = ['datacube*']
```

Comment:
Is the `*` intentional, so some other directory is generated at package build time and should be included in the release?

Reply:
I think it's to include the non-python resources loaded by code - various `alembic.ini`, `*.sql`, `*.yaml` etc files squirrelled away through the code-base?

Seems to be pretty standard in other pyproject.tomls though.